### PR TITLE
Securely deleting files

### PR DESCRIPTION
**Pull request by RutgerKe**
Added an option to securely delete files. This is achieved by using the
shred command (installed by default on ubuntu servers).

Tested by manually making a hard link of a file, when the option is
enabled the linked file is destroyed as well. When secure deletion is
disabled the linked file still has the original contents.

### CODE REVIEW

File: install/database.php [hunk 1]
```diff
@@ -23,7 +23,7 @@
 								  `timestamp` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP(),
 								  `uploader` varchar('.MAX_USER_CHARS.') NOT NULL,
 								  `expires` INT(1) NOT NULL default \'0\',
-								  `expiry_date` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP(),
+								  `expiry_date` TIMESTAMP NOT NULL DEFAULT \'0000-00-00 00:00:00\',
```
Comment (RutgerKe) [inline install/database.php]: I forgot all about changing this one, shouldn't have been in the same commit. But for my installation it does seem to be needed, the \'0000-00-00 00:00:00\' reintroduces issue #136 .

Comment (RutgerKe) [inline install/database.php]: Already figured out the problem, this is due to a configuration of mysql not allowing zero dates. 
Looking it up in the [mysql manual ](http://dev.mysql.com/doc/refman/5.7/en/sql-mode.html) does however say that it is default on mysql version 5.7.

**Comment by ignacionelson** [comment]
Thanks for the code! This is a great addition.
I will merge and then make one more change. We should make sure that unlink still is executed if shred fails and the file still exists.

**Comment by RutgerKe** [comment]
I don't think that is the right solution. If you want to securely delete a file you don't want it to (silently) unlink it if shred fails, that kind of defeats the purpose. The reasons why it would fail could also cause unlink to fail anyway.

Maybe an exec("shred --version") on the settings page could take away the doubt if it is installed? And only making it possible to enable it if that command succeeds? Or would it be too much overhead to execute that command everytime the settings page is opened?

**Comment by RutgerKe** [comment]
Sounds good! I'll probably have some time to try that at the end of this week, I won't be offended however if you can fix it before then ;)

Something else, I wasn't sure about the versioning? Especially in the core.update.php  (and the comment in site.options.php). Or is that something you will take care of at the time of merging?

**Comment by RutgerKe** [comment]
Took a little longer than expected, sorry for that. I'm not too sure if I updated correctly to the latest master version? I think I did...

**Comment by piorekf** [comment]
Just an FYI: On SSD drives, because of wear-leveling (cell remapping), shred is no more secure than normal rm.

**Comment by gaspolo** [comment]
old post , but i add this function to my projectS, editing php and database, so i have a new option near the file name, Wipe secure , and delete with "wipe" program the file from linux , and erase table from db. very easy.

**Comment by RutgerKe** [comment]
The FYI from @piorekf was spot on, we now run projectsend on a server with an SSD, so the feature is useless. I'm not planning on updating this or anything.

**Pull request closed without merging**